client/src/components/Charts/ManifestStatusPieChart/ManifestStatusPieChart.tsx

```diff
@@ -1,43 +1,297 @@
+          const activeAlphaColor = entry.color.slice(
+            entry.color.lastIndexOf(' ') + 1,
+            entry.color.length - 1
+          );
+          const activeLegend = activeAlphaColor === activeAlpha;
+          const baseStyle = { color: entry.color, paddingBottom: '2px' };
+          const spanStyle = activeLegend ? { ...baseStyle, borderBottom: `2px solid` } : baseStyle;
+
+          return (
+            <button
+              key={`item-${index}`}
+              onMouseEnter={() => handleMouseEnter(null, index)}
+              onMouseLeave={handleMouseLeave}
+              onClick={handleClick(dataEntry)}
+              className={`recharts-legend-item legend-item-${index}`}
+              style={{ display: 'inline-block', marginRight: '10px' }}
+            >
+              <svg
+                className="recharts-surface"
+                width="14"
+                height="14"
+                style={{ display: 'inline-block', verticalAlign: 'middle', marginRight: '4px' }}
+                viewBox="0 0 32 32"
+              >
+                <title></title>
+                <desc></desc>
+                <path
+                  stroke="none"
+                  fill={entry.color}
+                  d="M0,4h32v24h-32z"
+                  className="recharts-legend-icon"
+                ></path>
+              </svg>
+              <span className="recharts-legend-item-text" style={spanStyle}>
+                {entry.value}
+              </span>
+            </button>
+          );
+        })}
+      </ul>
+    </div>
+  );
+};
 
 export function ManifestStatusPieChart() {
+  const [activeIndex, setActiveIndex] = useState(-1);
+  const [chartRenderComplete, setChartRenderComplete] = useState(false);
   const navigate = useNavigate();
+
+  const handleMouseEnter = useCallback(
+    (_: any, index: number) => {
+      setActiveIndex(index);
+    },
+    [setActiveIndex]
+  );
+
+  const handleMouseLeave = useCallback(() => {
+    setActiveIndex(-1);
+  }, [setActiveIndex]);
+
+  const renderLabel = (props: any) => {
+    if (chartRenderComplete) {
+      return renderCustomLabel({ ...props, hover: false, activeIndex: activeIndex });
+    }
+  };
+
+  const handleClick = (entry: Entry) => () => {
+    navigate({
+      pathname: './manifest',
+      search: `?status=${entry.searchParam}`,
+    });
+  };
+
+  const animationDuration = chartRenderComplete ? 0 : 1000;
+
   return (
     <ResponsiveContainer minWidth={100} minHeight={300} height={'10%'}>
       <PieChart width={400} height={400}>
-        <Legend verticalAlign="bottom" height={36} />
+        <Legend
+          content={(props: any) =>
+            renderLegend({ ...props, handleMouseEnter, handleMouseLeave, handleClick })
+          }
+          verticalAlign="bottom"
+          height={36}
+        />
         <Pie
+          animationDuration={animationDuration}
+          onAnimationEnd={() => setChartRenderComplete(true)}
```

Comment:
@sheckathorne thanks for your patience. Just getting around to this. 

Sounds like you already pieced this part together from the PR. The callback passed to `onAnimationEnd` is triggering a state change (`chartRenderComplete`) which causes the entire component to rerender which strict mode doesn't like. 

We can simply axe everything related to that state. If we do that, the initial animation works correctly with strict mode enabled. I'll commit and push my changes for that part. 

There's a small hiccup, the labels are added before the animation is complete. Weirdly enough, if we comment out the `<Legend/>` element, that problem goes away. I can't explain that one yet.

Reply:
overall, I'm liking this direction this is headed. I'll stop work here and let you continue.

Reply:
Thanks for the cleanup and like you said, there is some definite unexpected behavior happening with the labels.

The problem with the labels is when you hover to activate a pie slice, then mouse away to white space, the labels don't reappear for 1500ms, equal to the default animation duration. You can combat this by setting `animationDuration` in the `Pie` element to 0, which also unfortunately removes the initial animation.

My counter to the above problem was the (admittedly unideal) `chartRenderComplete` state to set `animationDuration` to 0 after the initial animation was completed. However, problems with `onAnimationEnd` in strict mode is preventing the intended behavior.

I'm going to keep at this issue to see if there's some other solution I'm missing. Maybe the `Legend` element is the key.

Reply:
@sheckathorne I've debugged the issue. We just needed to use both `onAnimationEnd` and `onAnimationStart` to ensure that we're keeping state up to date.

This is ready for your review.